openedx/frontend-component-header
# feat: configureable header links

## Issue
https://github.com/openedx/platform-roadmap/issues/349

## Description
Makes header links, labels, and User name display more customizable using the configurations.

## Testing

- You can test this PR with the [Account MFE](https://github.com/openedx/frontend-app-account/) & [Learning MFE](https://github.com/openedx/frontend-app-learning/)
- Set these variables in .env.development of the Account & Learning MFE
  - `SHOW_FULLNAME=true`
  - `DASHBOARD_URL=<ANY URL>`
  - `EXTERNAL_ACCOUNT_PROFILE_URL=<ANY URL>`
  - `SHOW_SETTINGS_LABEL=true`
  - `ACCOUNT_SETTINGS_URL=http://localhost:1997`
- Use the local version of this header component and checkout this branch
- For header, run:
  - npm i
  - npm run build
- Now start the MFEs
- Header should reflect the configured changes.
- Also, test without the variables and the defaults should match the master branch.


## Screenshots
<img width="1736" alt="Screenshot 2024-03-29 at 5 34 39 PM" src="https://github.com/openedx/frontend-component-header/assets/52656433/291d23eb-6300-4287-a60e-efa24b054c41">
